pkg/reconciler/route/config/domain.go

```diff
@@ -31,9 +31,12 @@ import (
 const (
 	// DomainConfigName is the config map name for the domain configuration.
 	DomainConfigName = "config-domain"
+)
+
+var (
 	// DefaultDomain holds the domain that Route's live under by default
 	// when no label selector-based options apply.
-	DefaultDomain = "svc.cluster.local"
+	DefaultDomain = network.GetClusterDomainName()
```

Comment:
Does this return `cluster.local` or `svc.cluster.local` ? Cause technically kubernetes services have DNS records for `*.svc.[cluster-domain]`

So I worry that there might be a subtle bug that we're not catching in our tests because we are explicitly setting this value via an overlay 

Overlay 
https://github.com/knative/serving/blob/14ad4d1c6adb2479d4d64ced37f8b3570f963d55/test/config/ytt/core/overlay-cluster-local-override.yaml#L5-L16

default value for the overlay - https://github.com/knative/serving/blob/14ad4d1c6adb2479d4d64ced37f8b3570f963d55/test/config/ytt/values.yaml#L3-L5

Reply:
> Does this return cluster.local or svc.cluster.local ? Cause technically kubernetes services have DNS records for *.svc.[cluster-domain]

It returns `cluster.local` (hence why I needed to add `.svc` in the tests. See this unit test error from before: 
<img width="843" alt="image" src="https://github.com/knative/serving/assets/13594849/88e4a7a8-324c-456f-a9f2-c25719ae1383">

Reply:
I would expect that `svc` remain in this URL

https://github.com/knative/serving/pull/13964/files#diff-5ac543801229afd7c9de35d9e5ddef77987f6328550100a832b7836a7c974a2eR486-R487

Reply:
Since we're refering to K8s services - I think the domain is really 

```go
`svc.` + network.GetClusterDomainName()
```

Reply:
Thanks! That makes more sense.